job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql

```diff
@@ -36,13 +36,14 @@ BEGIN
         WITH RECURSIVE all_job_dependencies AS (
             SELECT partition_id, dependent_job_id, job_id
             FROM job_dependency
+            WHERE partition_id = in_partition_id AND job_id = in_job_id
             UNION ALL
             SELECT adj.partition_id, adj.dependent_job_id, jd.job_id
             FROM all_job_dependencies adj
             INNER JOIN job_dependency jd ON adj.partition_id = jd.partition_id AND adj.job_id = jd.dependent_job_id
+            WHERE adj.partition_id = in_partition_id
```

Comment:
Adding this `WHERE` allows the recursive clause to actually use the index on `job_dependency` instead of doing a seq scan of the table reach iteration.